pkg/builder/controller.go

```diff
@@ -228,24 +240,46 @@ func (blder *Builder) Build(r reconcile.Reconciler) (controller.Controller, erro
 	if r == nil {
 		return nil, fmt.Errorf("must provide a non-nil Reconciler")
 	}
-	if blder.mgr == nil {
+	if blder.mgr == nil || blder.cluster == nil {
 		return nil, fmt.Errorf("must provide a non-nil Manager")
 	}
 	if blder.forInput.err != nil {
 		return nil, blder.forInput.err
 	}
 
+	if err := blder.mgr.AddLogicalRunnableBuilder(func(name logical.Name, cl cluster.Cluster) (manager.Runnable, error) {
```

Comment:
`AddClusterRunnableBuilder` might be better yes. Potentially, we can require that a `cluster.Cluster` always comes with a `Name() logical.Name`?

Reply:
The other tricky bit to think about is the builder itself, right now every watch, handler, and controller is built against the manager's default cluster. That said, the Watch and Controller usually don't build the sources or handlers, but those are rather built in the builder, and then Watch is called on it.

It might be required to rework the controller builder to somehow track, when running in a fleet aware mode, how to build sources, handlers, and watches. Thoughts?